source/AI.cpp

```diff
@@ -103,7 +105,7 @@ namespace {
 				continue;
 			if(!escort->IsDisabled() && !escort->CanBeCarried()
 					&& escort->GetSystem() == ship.GetSystem()
-					&& escort->JumpFuel() && !escort->IsReadyToJump(true))
+					&& escort->JumpNavigation().JumpFuel() && !escort->IsReadyToJump(true))
```

Comment:
> This doesnt seem very clean, I was taught in the university that calling on methods on methods is something that should be avoided if possible

I don't particularly see why this would be an issue, so long as Navigation isn't giving out information it shouldn't. Ship only ever returns a const reference to its Navigation, and so Navigation can only ever tell outside classes exactly what would be told by having those methods remain on Ship and Ship just call Navigation. And in any case, this method of having functions that return objects that handle stuff for Ship isn't uncommon in the codebase; Armament, Cargo, and Attributes are such functions that are done this way.

> Do note I dont like ship having extra methods that much either, but if they're just methods that return another method they'll be pretty small anyway so it wont fill ship.cpp

Ship is a big class as is that needs trimmed. I'd rather take the gains where I can get them.

Reply:
I agree that we're not doing anything too complex with it currently so its fine, and repeating all methods would be annoying for sure, and I cant find any better ideas unfortunately other than handling the jumpnavigation internally instead of worrying about ships at all but that would be for another PR for sure
could also try to put it in a variable or smth when used a lot but I didnt see many instances of that its mostly spread out for now